tests/scripts/test-e2e-cluster.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash -e
+# Input env variables:
+# - PROVIDER - the LLM provider to be used during the test
+# - PROVIDER_KEY_PATH - path to a file containing the credentials to be used with the llm provider
+# - MODEL - name of the model to use during e2e testing
+
+# Script flow:
+# 1) Install OLS into a namespace on the cluster with valid config/api tokens and exposed via a route
+# 2) Setup a service account w/ permission to access OLS
+# 3) Wait for the ols api server to be available
+# 4) Invoke the test-e2e Makefile target
+
+
+DIR="${BASH_SOURCE%/*}"
+if [[ ! -d "$DIR" ]]; then DIR="$PWD"; fi
+. "$DIR/utils.sh"
+
+# temp directory for generated resource yamls
+TMPDIR=$(mktemp -d)
+
+# ARTIFACT_DIR is defined when running in a prow job, content
+# in this location is automatically collected at the end of the test job
+ARTIFACT_DIR=${ARTIFACT_DIR:-$TMPDIR}
+
```

Comment:
It seems like we only need the TMPDIR when running this script outside of prow. If that's the case, would it be clearer to write this as
```
if [ -n "$ARTIFACT_DIR" ]; then
    readonly ARTIFACT_DIR=$(mktemp -d)
    trap 'rm -rf $ARTIFACT_DIR' EXIT
fi

```
?

Reply:
actually i think you mean `-z` and not `-n`